### PR TITLE
clarify use of responseType attribute

### LINKED ISSUE #8808: OIDC RP does notHTTP Auth header as containing a valid OIDC id_token
TS002246300
Our team is developing an application that should be able to accept OAuth 2.0 or OpenConnect ID tokens as Bearer tokens on the HTTP Authorization header to authenticate to the application hosted on WebSphere Liberty.

In addition, if the end user has not previously authenticated to our application, they are redirected to a Liberty-based Open ID Connect Provider for authentication. Then after they have authenticated, they are redirected back to our application as an OIDC RP.

In this latter scenario, if the openidConnectClient configuration sets the inboundPropagation attribute to "supported', this web-based authentication scenario does not work. Even though they have authenticated with the OIDC OP, they cannot access our application. The following error message appears in our application's log:

[5/13/19 11:42:59:273 PDT] 00044cdb .ibm.ws.security.openidconnect.client.jose4j.util.Jose4jUtil E CWWKS1712E: The OpenID Connect client [contentServicesUms] did not receive an ID token from the OpenID Connect provider [https://company.com:9443/oidc/endpoint/ums/token].

Below is the openidConnectClient configuration for our application:

```
<openidConnectClient id="ContentServicesUms"
scope="openid profile email"
responseType="id_token token"
clientId="myId"
clientSecret="mySecret"
issuerIdentifier="https://company.com:9443/oidc/endpoint/ums"
audiences="contentServicesUms"
validationMethod="introspect"
inboundPropagation="supported"
validationEndpointUrl="https://company.com:9443/oidc/endpoint/ums/introspect"
authorizationEndpointUrl="https://company.com:9443/oidc/endpoint/ums/authorize"
tokenEndpointUrl="https://company.com:9443/oidc/endpoint/ums/token">
</openidConnectClient>
```

If I change the inboundPropagation attribute to "none", then the web based authentication works, but the pre-existing OAuth 2.0 token and OIDC id_token scenarios no longer works when passed in as Authorization Bearer tokens.

I also noticed in the web based authentication flow, a cookie is set by the Liberty OIDC OP called "WASOidcCode" that contains the OIDC id_token.

When the responseType is changed to **code**, the scenario works.

Shouldn't a Liberty OIDC RP be able to use recognize this cookie as containing a valid OIDC id_token when using implicit flow (**responseType="id_token token"**)?

### PR DESCRIPTION
**Pull request by brutif**
resolves #8808

### CODE REVIEW

**Comment by brutif** [comment]
#build

File: dev/com.ibm.ws.security.openidconnect.client/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -105,13 +105,13 @@ jwkClientSecret=JWK client password
...
 grantType=Grant type
-grantType.desc=Specifies the grant type to use for this client.
+grantType.desc=Specifies the grant type to use for this client. Use of the responseType attribute is preferred instead.
 
 grantType.authorization_code=Authorization code grant type
 grantType.implicit=Implicit grant type
 
 responseType=Response type
-responseType.desc=Specifies the required response type for this client.
+responseType.desc=Specifies the response requested from the provider, either an authorization code or implicit flow tokens.
```
Comment (ayoho) [inline dev/com.ibm.ws.security.openidconnect.client/resources/OSGI-INF/l10n/metatype.properties]: Would probably prefer to say "Specifies the response **type** that is requested from the provider"

File: dev/com.ibm.ws.security.openidconnect.client/resources/OSGI-INF/l10n/metatype.properties [hunk 2]
```diff
@@ -105,13 +105,13 @@ jwkClientSecret=JWK client password
 jwkClientSecret.desc=Specifies the client password to include in the basic authentication scheme of the JWK request.
 
 grantType=Grant type
-grantType.desc=Specifies the grant type to use for this client.
+grantType.desc=Specifies the grant type to use for this client. Use of the responseType attribute is preferred instead.
```
Comment (dmuelle) [inline dev/com.ibm.ws.security.openidconnect.client/resources/OSGI-INF/l10n/metatype.properties]: Instead of what? The inout that generated this meesage?

**Comment by andrea-pichler** [comment]
No message review requested from L2

**Pull request merged**